pkg/controllers/clusterresourceplacement/placement_status_test.go

```diff
@@ -909,49 +968,1282 @@ func TestSetPlacementStatus(t *testing.T) {
+						fleetv1beta1.NumberOfResourceSnapshotsAnnotation: "1",
+					},
+				},
+			},
+			clusterResourceBindings: []fleetv1beta1.ClusterResourceBinding{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "binding-1",
+						Labels: map[string]string{
+							fleetv1beta1.CRPTrackingLabel: testName,
+						},
+						Generation: 1,
+					},
+					Spec: fleetv1beta1.ResourceBindingSpec{
+						State:                        fleetv1beta1.BindingStateBound,
+						ResourceSnapshotName:         fmt.Sprintf(fleetv1beta1.ResourceSnapshotNameFmt, testName, 0),
+						SchedulingPolicySnapshotName: fmt.Sprintf(fleetv1beta1.PolicySnapshotNameFmt, testName, 0),
+						TargetCluster:                cluster1Name,
+					},
+					Status: fleetv1beta1.ResourceBindingStatus{
+						Conditions: []metav1.Condition{
+							{
+								Status:             metav1.ConditionFalse,
+								Type:               string(fleetv1beta1.ResourceBindingBound),
+								Reason:             "resourceBindingTrue",
+								ObservedGeneration: 1,
+								LastTransitionTime: oldTransitionTime,
+							},
+						},
+					},
+				},
+			},
+			wantStatus: &fleetv1beta1.ClusterResourcePlacementStatus{
+				SelectedResources: selectedResources,
+				Conditions: []metav1.Condition{
+					{
+						Status:             metav1.ConditionUnknown,
+						Type:               string(fleetv1beta1.ClusterResourcePlacementAppliedConditionType),
+						Reason:             ApplyPendingReason,
+						ObservedGeneration: crpGeneration,
+						LastTransitionTime: oldTransitionTime,
+					},
+					{
+						Status:             metav1.ConditionTrue,
+						Type:               string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType),
+						Reason:             "Scheduled",
+						ObservedGeneration: crpGeneration,
+						LastTransitionTime: oldTransitionTime,
+					},
+					{
+						Status:             metav1.ConditionFalse,
+						Type:               string(fleetv1beta1.ClusterResourcePlacementSynchronizedConditionType),
+						Reason:             SynchronizePendingReason,
+						ObservedGeneration: crpGeneration,
+						LastTransitionTime: oldTransitionTime,
+					},
+				},
+				PlacementStatuses: []fleetv1beta1.ResourcePlacementStatus{
+					{
+						ClusterName:              cluster1Name,
+						FailedResourcePlacements: []fleetv1beta1.FailedResourcePlacement{},
+						Conditions: []metav1.Condition{
+							{
+								Status:             metav1.ConditionUnknown,
+								Type:               string(fleetv1beta1.ResourcesAppliedConditionType),
+								Reason:             ResourceApplyPendingReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionFalse,
+								Type:               string(fleetv1beta1.ResourceWorkSynchronizedConditionType),
+								Reason:             WorkSynchronizeFailedReason,
+								ObservedGeneration: crpGeneration,
```

Comment:
should this still be crpGeneration? This could be work/resource generation instead. The CRP generation doesn't tell the user much about the apply or sync condition as those can happen behind CRP's. back.

Reply:
yeah, that's true. But the condition is for the CRP resource itself. The new field resourceSnapshots of status should solve this issue.